ludwig/data/preprocessing.py

```diff
@@ -1414,6 +1431,11 @@ def build_data(
         # Need to run this again here as cast_columns may have introduced new missing values
         handle_missing_values(input_cols, feature_config, preprocessing_parameters, backend)
 
+        # For features that support it, we perform outlier removal here using metadata computed on the full dataset
+        handle_outliers(
+            input_cols, feature_config, preprocessing_parameters, training_set_metadata[feature_config[NAME]], backend
+        )
+
```

Comment:
QQ - when you say `full dataset`, do you mean each of the splits individually or just the train set?

Reply:
nvm silly question, it's training set metadata

Reply:
At this point it's always the combined train + val + test dataset. Splitting happens at the end.